operator/cmd/sidecar/sidecar.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package sidecar
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+
+	"github.com/redpanda-data/redpanda-operator/operator/internal/decommissioning"
+)
+
+var schemes = []func(s *runtime.Scheme) error{
+	clientgoscheme.AddToScheme,
+}
+
+func Command() *cobra.Command {
+	var (
+		metricsAddr                string
+		probeAddr                  string
+		pprofAddr                  string
+		clusterNamespace           string
+		clusterName                string
+		decommissionRequeueTimeout time.Duration
+		decommissionVoteInterval   time.Duration
+		decommissionMaxVoteCount   int
+		redpandaYAMLPath           string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "run",
+		Short: "Run the redpanda sidecar",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			return Run(
+				ctx,
+				metricsAddr,
+				probeAddr,
+				pprofAddr,
+				clusterNamespace,
+				clusterName,
+				decommissionRequeueTimeout,
+				decommissionVoteInterval,
+				decommissionMaxVoteCount,
+				redpandaYAMLPath,
+			)
+		},
+	}
+
+	cmd.Flags().StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	cmd.Flags().StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	cmd.Flags().StringVar(&pprofAddr, "pprof-bind-address", ":8082", "The address the metric endpoint binds to.")
+	cmd.Flags().StringVar(&clusterNamespace, "cluster-namespace", "", "The namespace of the cluster that this sidecar manages.")
+	cmd.Flags().StringVar(&clusterName, "cluster-name", "", "The name of the cluster that this sidecar manages.")
+	cmd.Flags().DurationVar(&decommissionRequeueTimeout, "decommission-requeue-timeout", 10*time.Second, "The time period to wait before rechecking a broker that is being decommissioned.")
+	cmd.Flags().DurationVar(&decommissionVoteInterval, "decommission-vote-interval", 30*time.Second, "The time period between incrementing decommission vote counts since the last decommission conditions were met.")
+	cmd.Flags().IntVar(&decommissionMaxVoteCount, "decommission-vote-count", 2, "The number of times that a vote must be tallied when a resource meets decommission conditions for it to actually be decommissioned.")
+	cmd.Flags().StringVar(&redpandaYAMLPath, "redpanda-yaml", "/etc/redpanda/redpanda.yaml", "Path to redpanda.yaml")
```

Comment:
Want to give a bit more description about what the redpanda.yaml is used for in this context? (Given that it's quite broad...)